src/handshakestate.rs

```diff
@@ -506,6 +508,17 @@ impl HandshakeState {
         self.my_turn
     }
 
+    /// Perform the split calculation and return the resulting keys.
+    ///
+    /// This returns raw key material so it should be used with care. The "risky-raw-split"
+    /// feature has to be enabled to use this function.
+    #[cfg(feature = "risky-raw-split")]
+    pub fn dangerously_get_raw_split(&mut self) -> (Vec<u8>, Vec<u8>) {
```

Comment:
Noise currently only ever uses 32-byte keys, so it may make more sense here to actually return `([u8; CIPHERKEYLEN], [u8; CIPHERKEYLEN])` instead of doing a heap allocation here.